KSP: Introspection generated duplication controller when you use interfaces and `@Controller` annotaion  exists in interface and implementation class

### Expected Behavior

Sample code:

Interface:
```kotlin
@Controller
interface MyEntityControllerInterface {

    @Post("/sendModelWithMapProperty")
    fun sendModelWithMapProperty(
            @Body @NotNull @Valid modelWithMapProperty: ModelWithMapProperty
    ): Mono<ModelWithMapProperty>
}
```
Implementation:
```kotlin
@Controller
open class MyEntityController(
        @Client("local") val httpClient: HttpClient
) : MyEntityControllerInterface {

    override fun sendModelWithMapProperty(modelWithMapProperty: ModelWithMapProperty): Mono<ModelWithMapProperty> {
        println("GOT IT!!!!")
        return Mono.just(modelWithMapProperty)
    }
}
```

KAPT introspection generated these classes:
![изображение](https://github.com/micronaut-projects/micronaut-core/assets/48474279/4f9a4960-53f5-4cd0-9145-e4b0a51ba207)

KSP introspection generated these classes:
![изображение](https://github.com/micronaut-projects/micronaut-core/assets/48474279/46295f45-7e3d-4c56-833f-c809181d198c)

As you see, KAPT ignore interface and this is correct, but KSP identificate interface as different bean, and after it we have two controllers with same path.

In tests you have this error:
```java
{"_links":{"self":[{"href":"/api/sendModelWithMapProperty","templated":false}]},"_embedded":{"errors":[{"message":"More than 1 route matched the incoming request. The following routes matched /api/sendModelWithMapProperty: POST - /api/sendModelWithMapProperty, POST - /api/sendModelWithMapProperty"}]},"message":"Bad Request"}
```

When you will try to start application, you will see this excpetion:
```java
17:29:23.500 ERROR [main] i.m.runtime.Micronaut [Micronaut.java:337] - Error starting Micronaut server: Bean definition [com.micronaut.bug.controller.$MyEntityControllerInterface$Definition$Intercepted] could not be loaded: Error instantiating bean of type  [com.micronaut.bug.controller.$MyEntityControllerInterface$Definition$Intercepted]

Message: com.micronaut.bug.controller.MyEntityControllerInterface
Path Taken: new $MyEntityControllerInterface$Definition$Intercepted(BeanResolutionContext $beanResolutionContext,BeanContext $beanContext,Qualifier $qualifier,List $interceptors)
io.micronaut.context.exceptions.BeanInstantiationException: Bean definition [com.micronaut.bug.controller.$MyEntityControllerInterface$Definition$Intercepted] could not be loaded: Error instantiating bean of type  [com.micronaut.bug.controller.$MyEntityControllerInterface$Definition$Intercepted]

Message: com.micronaut.bug.controller.MyEntityControllerInterface
Path Taken: new $MyEntityControllerInterface$Definition$Intercepted(BeanResolutionContext $beanResolutionContext,BeanContext $beanContext,Qualifier $qualifier,List $interceptors)
	at io.micronaut.context.DefaultBeanContext.initializeContext(DefaultBeanContext.java:1980)
	at io.micronaut.context.DefaultApplicationContext.initializeContext(DefaultApplicationContext.java:291)
	at io.micronaut.context.DefaultBeanContext.readAllBeanDefinitionClasses(DefaultBeanContext.java:3322)
	at io.micronaut.context.DefaultBeanContext.finalizeConfiguration(DefaultBeanContext.java:3677)
	at io.micronaut.context.DefaultBeanContext.start(DefaultBeanContext.java:361)
	at io.micronaut.context.DefaultApplicationContext.start(DefaultApplicationContext.java:199)
	at io.micronaut.runtime.Micronaut.start(Micronaut.java:73)
	at io.micronaut.runtime.Micronaut.run(Micronaut.java:322)
	at io.micronaut.runtime.Micronaut.run(Micronaut.java:308)
	at com.micronaut.bug.Application$Companion.main(Application.kt:25)
	at com.micronaut.bug.Application.main(Application.kt)
Caused by: io.micronaut.context.exceptions.BeanInstantiationException: Error instantiating bean of type  [com.micronaut.bug.controller.$MyEntityControllerInterface$Definition$Intercepted]

Message: com.micronaut.bug.controller.MyEntityControllerInterface
Path Taken: new $MyEntityControllerInterface$Definition$Intercepted(BeanResolutionContext $beanResolutionContext,BeanContext $beanContext,Qualifier $qualifier,List $interceptors)
	at io.micronaut.context.DefaultBeanContext.resolveByBeanFactory(DefaultBeanContext.java:2324)
	at io.micronaut.context.DefaultBeanContext.doCreateBean(DefaultBeanContext.java:2279)
	at io.micronaut.context.DefaultBeanContext.doCreateBean(DefaultBeanContext.java:2291)
	at io.micronaut.context.DefaultBeanContext.createRegistration(DefaultBeanContext.java:3054)
	at io.micronaut.context.SingletonScope.getOrCreate(SingletonScope.java:80)
	at io.micronaut.context.DefaultBeanContext.findOrCreateSingletonBeanRegistration(DefaultBeanContext.java:2956)
	at io.micronaut.context.DefaultBeanContext.resolveBeanRegistration(DefaultBeanContext.java:2917)
	at io.micronaut.context.DefaultBeanContext.getProxyTargetBean(DefaultBeanContext.java:1489)
	at com.micronaut.bug.controller.$MyEntityControllerInterface$Definition$Intercepted.<init>(Unknown Source)
	at com.micronaut.bug.controller.$MyEntityControllerInterface$Definition$Intercepted$Definition.instantiate(Unknown Source)
	at io.micronaut.context.DefaultBeanContext.resolveByBeanFactory(DefaultBeanContext.java:2309)
	at io.micronaut.context.DefaultBeanContext.doCreateBean(DefaultBeanContext.java:2279)
	at io.micronaut.context.DefaultBeanContext.doCreateBean(DefaultBeanContext.java:2291)
	at io.micronaut.context.DefaultBeanContext.createRegistration(DefaultBeanContext.java:3054)
	at io.micronaut.context.SingletonScope.getOrCreate(SingletonScope.java:80)
	at io.micronaut.context.DefaultBeanContext.findOrCreateSingletonBeanRegistration(DefaultBeanContext.java:2956)
	at io.micronaut.context.DefaultBeanContext.initializeEagerBean(DefaultBeanContext.java:2676)
	at io.micronaut.context.DefaultBeanContext.initializeContext(DefaultBeanContext.java:1974)
	... 10 common frames omitted
Caused by: java.lang.InstantiationError: com.micronaut.bug.controller.MyEntityControllerInterface
	at com.micronaut.bug.controller.$MyEntityControllerInterface$Definition.instantiate(Unknown Source)
	at io.micronaut.context.DefaultBeanContext.resolveByBeanFactory(DefaultBeanContext.java:2309)
	... 27 common frames omitted
```
### Version

4.2.3

--- a/core-processor/src/main/java/io/micronaut/inject/processing/BeanDefinitionCreatorFactory.java
+++ b/core-processor/src/main/java/io/micronaut/inject/processing/BeanDefinitionCreatorFactory.java
@@ -66,6 +66,9 @@ public abstract class BeanDefinitionCreatorFactory {
             }
             return new AopIntroductionProxySupportedBeanElementCreator(classElement, visitorContext, false);
         }
+        if (classElement.isInterface()) {
+            return Collections::emptyList;
+        }
         // NOTE: In Micronaut 3 abstract classes are allowed to be beans, but are not pickup to be beans just by having methods or fields with @Inject
         if (isDeclaredBean(classElement) || (!isAbstract && (containsInjectMethod(classElement) || containsInjectField(classElement)))) {
             if (classElement.hasStereotype("groovy.lang.Singleton")) {

--- a/inject-java/src/test/groovy/io/micronaut/inject/beans/BeanDefinitionSpec.groovy
+++ b/inject-java/src/test/groovy/io/micronaut/inject/beans/BeanDefinitionSpec.groovy
@@ -757,4 +757,36 @@ so we mark it as overridden
         cleanup:
             ctx.close()
     }
+
+    void "test interface bean"() {
+        given:
+            def definition = buildBeanDefinition('test.MyEntityControllerInterface', '''
+package test;
+
+import io.micronaut.http.annotation.Controller;
+
+@Controller
+interface MyEntityControllerInterface {
+}
+''')
+
+        expect:
+            definition == null
+    }
+
+    void "test interface bean 2"() {
+        given:
+            def definition = buildBeanDefinition('test.MyEntityControllerInterface', '''
+package test;
+
+import jakarta.inject.Singleton;
+
+@Singleton
+interface MyEntityControllerInterface {
+}
+''')
+
+        expect:
+            definition == null
+    }
 }

--- a/inject-kotlin/src/test/groovy/io/micronaut/kotlin/processing/beans/BeanDefinitionSpec.groovy
+++ b/inject-kotlin/src/test/groovy/io/micronaut/kotlin/processing/beans/BeanDefinitionSpec.groovy
@@ -130,6 +130,38 @@ class F
         definition.injectedFields.size() == 1
     }
 
+    void "test interface bean"() {
+        given:
+        def definition = KotlinCompiler.buildBeanDefinition('test.MyEntityControllerInterface', '''
+package test
+
+import io.micronaut.http.annotation.Controller
+
+@Controller
+interface MyEntityControllerInterface {
+}
+''')
+
+        expect:
+            definition == null
+    }
+
+    void "test interface bean 2"() {
+        given:
+        def definition = KotlinCompiler.buildBeanDefinition('test.MyEntityControllerInterface', '''
+package test
+
+import jakarta.inject.Singleton
+
+@Singleton
+interface MyEntityControllerInterface {
+}
+''')
+
+        expect:
+            definition == null
+    }
+
     @PendingFeature(reason = "difficult to achieve with current design without a significant rewrite or how native properties are handled")
     void "test injection order for inheritance"() {
         given:

